src/test/java/com/stormeye/evaluation/DeployStepDefinitions.java

```diff
@@ -0,0 +1,195 @@
+
+        final Ed25519PrivateKey senderKey = new Ed25519PrivateKey();
+        final Ed25519PublicKey receiverKey = new Ed25519PublicKey();
+
+        senderKey.readPrivateKey(AssetUtils.getUserKeyAsset(1, senderId, "secret_key.pem").getFile());
+        receiverKey.readPublicKey(AssetUtils.getUserKeyAsset(1, receiverId, "public_key.pem").getFile());
+
+        parameterMap.put("senderKey", senderKey);
+        parameterMap.put("receiverKey", receiverKey);
+    }
+
+    @And("the deploy has a ttl of {int}m")
+    public void theDeployHasATtlOfM(int ttlMinutes) {
+
+        logger.info("And the deploy has a ttl of {}m", ttlMinutes);
+
+        parameterMap.put("ttl", Ttl.builder().ttl(ttlMinutes + "m").build());
+    }
+
+    @And("the transfer amount is {long}")
+    public void theTransferAmountIs(long amount) {
+
+        logger.info("And the transfer amount is {}", amount);
+
+        parameterMap.put("transferAmount", BigInteger.valueOf(amount));
+    }
+
+    @And("the transfer gas price is {long}")
+    public void theTransferPriceIs(long price) {
+
+        logger.info("And the transfer gas price is {}", price);
+
+        parameterMap.put("gasPrice", price);
+    }
+
+    @When("the deploy is put on chain {string}")
+    public void theDeployIsPut(final String chainName) throws Exception {
+
+        logger.info("When the deploy is put on chain {}", chainName);
+
+        final Deploy deploy = CasperTransferHelper.buildTransferDeploy(
+                parameterMap.get("senderKey"),
+                PublicKey.fromAbstractPublicKey(parameterMap.get("receiverKey")),
+                parameterMap.get("transferAmount"),
+                chainName,
+                Math.abs(new Random().nextLong()),
+                BigInteger.valueOf(100000000L),
+                parameterMap.get("gasPrice"),
+                parameterMap.get("ttl"),
+                new Date(),
+                new ArrayList<>());
+
+
+        final CasperService casperService = CasperClientProvider.getInstance().getCasperService();
+
+        parameterMap.put("deployResult", casperService.putDeploy(deploy));
+    }
+
+
+    @Then("the deploy response contains a valid deploy hash of length {int} and an API version {string}")
+    public void theValidDeployHashIsReturned(final int hashLength, final String apiVersion) {
+
+        logger.info("Then the deploy response contains a valid deploy hash of length {} and an API version {}", hashLength, apiVersion);
+
+        DeployResult deployResult = parameterMap.get("deployResult");
+        assertThat(deployResult, is(notNullValue()));
+        assertThat(deployResult.getDeployHash(), is(notNullValue()));
+        assertThat(deployResult.getDeployHash().length(), is((hashLength)));
+        assertThat(deployResult.getApiVersion(), is((apiVersion)));
+
+        logger.info("deployResult.getDeployHash() {}", deployResult.getDeployHash());
```

Comment:
we should get the block from nctl as well and do a compare
this can be for later, I'll work on a block compare class